src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -349,6 +350,15 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         .collect(Collectors.toList());
   }
 
+  public UUID getFirstSourceDatasetIdFromSnapshotId(
```

Comment:
I would instead call this `getSourceDatasetIdFromSnapshotId()`, because I think the caller doesn't want the "first" source, it wants _the_ source. It happens that this is the first one, but the caller doesn't need to know that.

Reply:
I'm specifying that to follow the getFirstSnapshotSource() convention to make it clear that we need to change this one day when we allow more than one source dataset per snapshot. You're right in that "first" isn't so important, but it's the convention that was picked for the other call.

Reply:
I think separation of concerns applies here too. Regardless of what we support in the future, the API today returns "the" dataset for a snapshot. If we needed to support more than one, we'd need to change this API to return an array and the caller to handle that case. In the meantime, a caller doesn't need to know where "the" dataset is stored. It could be the first, last, second, etc. Adding `first` to the name exposes an implementation detail that the caller doesn't need to be aware of.